### ISSUE TITLE
ClickHouse: identifiers beginning with a digit (e.g. 30_day_cm3) are unparsable

### ISSUE THREAD

**Issue by friedrich123456789** [sqlfluff/sqlfluff#8035]
### What happened

Unquoted identifiers that begin with a digit but contain letters
(`3s_video_views`, `30_day_cm3`) are reported as unparsable — both as column
references and as aliases.

### Expected behaviour

They parse as identifiers.

Note: ClickHouse's syntax docs describe the stricter regex
`^[a-zA-Z_][0-9a-zA-Z_]*$` for non-quoted identifiers, but in practice the
ClickHouse parser accepts digit-leading names unquoted (verified on 26.4.2 —
see below). So this may be best addressed as relaxed lexing and/or a config
opt-in rather than unconditionally; happy to follow whatever policy the team
prefers.

### Observed behaviour

```
SELECT 3s_video_views FROM t      ->  L1:P8  PRS: Found unparsable section: '3s_video_views'
SELECT 1 AS 30_day_cm3            ->  L1:P13 PRS: Found unparsable section: 'AS 30_day_cm3'
```

### How to reproduce

```sql
SELECT 3s_video_views FROM t
SELECT 1 AS 30_day_cm3
```

```bash
sqlfluff parse --dialect clickhouse -
```

Verified on ClickHouse 26.4.2: `SELECT 3s_video_views FROM t`,
`SELECT 1 AS 3s_video_views`, `SELECT 1 AS 30_day_cm3` all accepted unquoted.
Docs: https://clickhouse.com/docs/sql-reference/syntax#identifiers

### Dialect

clickhouse

### Version

sqlfluff 4.2.2 (latest release), Python 3.12; also applies to `main`.

### Configuration

None — default rules, `--dialect clickhouse`.

**Comment by peterbud** [comment]
Hi @friedrich123456789  thanks for opening this issue.

I have also checked here the clickhouse docs: https://clickhouse.com/docs/sql-reference/syntax#identifiers
It's clearly defining that 
> Non-quoted identifiers must match the regex `^[a-zA-Z_][0-9a-zA-Z_]*$`

Based on this idetifiers cannot start with number IMHO.

Thoughts?

**Comment by peterbud** [comment]
Closing due to lack of response
